demonstrations/learning2learn.py

```diff
@@ -0,0 +1,1205 @@
+
+By building on results from *meta-learning* literature in Classical ML,
+authors in [#l2l]_ propose to use a Recurrent Neural Network (RNN)
+as a black-box controller to optimize the tunable parameters of
+variational quantum algorithms, as shown in the figure below. The cost
+function used is the expectation value :math:`\langle H \rangle_{\boldsymbol{\theta}} = \langle \psi_{\boldsymbol{\theta}} | H | \psi_{\boldsymbol{\theta}}\rangle`
+of a Hamiltonian :math:`H` with respect to the parametrized state
+:math:`|\psi_\boldsymbol{\theta}\rangle` evolved by applying the variational quantum circuit to the zero state.
+
+.. figure:: ../demonstrations/learning2learn/HybridLSTM.png
+    :align: center
+    :width: 100%
+
+Given parameters :math:`\boldsymbol{\theta}_{t-1}` of the variational quantum circuit,
+the cost function :math:`y_{t-1}`, and the hidden state of the
+classical network :math:`\boldsymbol{h}_{t-1}` at the previous time step, the
+classical computer (the Recurrent Neural Network) proposes a new
+guess for the parameters :math:`\boldsymbol{\theta}_t`, which are
+then fed into the quantum computer (the QAOA circuit) to evaluate the
+cost function :math:`y_t`. By repeating this cycle a few times, and
+by training the weights of the recurrent neural network to minimize
+the loss function :math:`y_t`, a good initialization heuristic is
+found for the parameters :math:`\boldsymbol{\theta}` of the variational
+quantum circuit.
+
+At a given iteration, the RNN receives as input the previous cost
+function :math:`y_t` evaluated on the quantum computer, where
+:math:`y_t` is the estimate of :math:`\langle H\rangle_{t}`, as well as
+the parameters :math:`\boldsymbol{\theta}_t` for which the variational
+circuit was evaluated. The RNN at this time step also receives
+information stored in its internal hidden state from the previous time
+step :math:`\boldsymbol{h}_t`. The RNN itself has trainable parameters :math:`\phi`,
+and hence it applies the parametrized mapping:
+
+.. math::  \boldsymbol{h}_{t+1}, \boldsymbol{\theta}_{t+1} = \text{RNN}_{\phi}(\boldsymbol{h}_{t}, \boldsymbol{\theta}_{t}, y_{t})
+
+which generates a new suggestion for the variational parameters as well
+as a new internal state. Upon training the weights :math:`\phi`, the RNN
+eventually learns a good heuristic to suggest optimal parameters for the
+quantum circuit.
+
+VQAs in focus: QAOA for MaxCut
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are multiple VQAs for which this hybrid training routine could
+be used, some of them directly analyzed in [#l2l]_. In the
+following, we will only focus on one of such examples, that is the use
+of the QAOA (Quantum Approximate Optimization Algorithm) for solving
+the MaxCut problem for graphs. Thus, referring to the picture above,
+the shape of the variational circuit is the one dictated by the QAOA
+ansatz, and such a quantum circuit is used to evaluate the cost
+Hamiltonian :math:`H` of the MaxCut problem of some graphs.
```

Comment:
```suggestion
Hamiltonian :math:`H` of the MaxCut problem.
```
And add a link to a site explaining MaxCut, e.g., wikipedia